make_mock_json.py

```diff
@@ -0,0 +1,310 @@
+    }
+}
+DEFAULT_DIR = "tests/mock_jsons"
+
+
+def canonize_data(build_data, build_name=TEST_BUILD, build_status=None):
+    build_data["status"]["config"] = {
+        "kind": "BuildConfig",
+        "name": build_name,
+        "namespace": "default",
+    }
+    build_data["metadata"]["name"] = build_name
+    if build_status:
+        build_data["status"]["phase"] = build_status
+        try:
+            del build_data["status"]["logSnippet"]
+            del build_data["status"]["message"]
+        except KeyError:
+            pass
+    return build_data
+
+
+def find_or_make_dir(dir_name):
+    try:
+        os.stat(dir_name)
+    except OSError:
+        try:
+            os.makedirs(dir_name)
+        except OSError:
+            print("cannot open or create {}".format(dir_name))
+            raise
+
+
+class Mock_Creator(object):
+    def __init__(self):
+        parser = argparse.ArgumentParser(description="osbs test harness mock JSON creator")
+
+        parser.add_argument("--config", action='store', metavar="PATH",
+                            help="path to configuration file", default=DEFAULT_CONFIGURATION_FILE)
+        parser.add_argument("--instance", "-i", action='store', metavar="SECTION_NAME",
+                            help="section within config for requested instance",
+                            default="stage")
+        parser.add_argument("--user", action='store',
+                            help="name of user to use for Basic Authentication in OSBS")
+        parser.add_argument("--password", action='store',
+                            help="password to use for Basic Authentication in OSBS")
+        parser.add_argument("--mock-dir", metavar="DIR", action="store", default=DEFAULT_DIR,
+                            help="mock JSON responses are stored in DIR")
+        parser.add_argument("--os_version", metavar="OS_VER", action="store", default=OS_VERSION,
+                            help="OpenShift version of the mock JSONs")
+        args = parser.parse_args()
+        self.user = args.user
+        self.password = args.password
+
+        mock_path = args.mock_dir
+        self.mock_dir = "/".join([mock_path, args.os_version])
+        find_or_make_dir(self.mock_dir)
+
+        self.capture_dir = os.tmpnam()
+        find_or_make_dir(self.capture_dir)
+
+        args.git_url = "https://github.com/TomasTomecek/docker-hello-world.git"
+        args.git_branch = "master"
+        args.git_commit = "HEAD"
+
+        os_conf = Configuration(conf_file=args.config, conf_section=args.instance, cli_args=args)
+        build_conf = Configuration(conf_file=args.config, conf_section=args.instance,
+                                   cli_args=args)
+
+        set_logging(level=logging.INFO)
+
+        self.osbs = OSBS(os_conf, build_conf)
+        setup_json_capture(self.osbs, os_conf, self.capture_dir)
+
+    def comp_write(self, out_name, out_data):
+        outs = json.dumps(out_data)
```

Comment:
We're calling `json.dumps()` on this again in the `except` clause, and it isn't used anywhere else.

Reply:
Fixed.